controllers/apply/form-router-next/submission.js

```diff
@@ -0,0 +1,181 @@
+'use strict';
+const express = require('express');
+const path = require('path');
+const unset = require('lodash/unset');
+
+const {
+    PendingApplication,
+    SubmittedApplication
+} = require('../../../db/models');
+
+const appData = require('../../../common/appData');
+const logger = require('../../../common/logger').child({
+    service: 'salesforce'
+});
+
+const salesforceService = require('./lib/salesforce');
+const { buildMultipartData } = require('./lib/file-uploads');
+
+module.exports = function(
+    formId,
+    formBuilder,
+    confirmationBuilder,
+    currentlyEditingSessionKey,
+    enableSalesforceConnector
+) {
+    const router = express.Router();
+
+    /**
+     * Route: Submission
+     */
+    router.post('/', async (req, res, next) => {
+        const { currentApplication, currentApplicationData } = res.locals;
+
+        const form = formBuilder({
+            locale: req.i18n.getLocale(),
+            data: currentApplicationData
+        });
+
+        if (form.progress.isComplete === false) {
+            return res.redirect(req.baseUrl);
+        }
+
+        function renderConfirmation() {
+            unset(req.session, `${currentlyEditingSessionKey()}`);
+            req.session.save(function() {
+                const confirmation = confirmationBuilder({
+                    locale: req.i18n.getLocale(),
+                    data: currentApplicationData
+                });
+
+                logger.info('Submission successful');
+                res.render(path.resolve(__dirname, './views/confirmation'), {
+                    title: confirmation.title,
+                    confirmation: confirmation,
+                    form: form
+                });
+            });
+        }
+
+        /**
+         * Fake submission for get advice form
+         * @TODO: SubmittedApplication assumes project title
+         */
+        if (formId === 'get-advice' && appData.isNotProduction) {
```

Comment:
Currently the submitted application model requires an application so needs reworking. For now I've bypassed _all_ saving for the get advice form.